multimedia/ffmpeg7/files/patch-libavdevice-avfoundation.diff

```diff
@@ -0,0 +1,13 @@
+--- libavdevice/avfoundation.m
++++ libavdevice/avfoundation.m
+@@ -763,6 +763,10 @@ static int get_audio_config(AVFormatContext *s)
+     return 0;
+ }
+
++#if ((TARGET_OS_IPHONE && __IPHONE_OS_VERSION_MAX_ALLOWED < 110000) || (TARGET_OS_OSX && __MAC_OS_X_VERSION_MAX_ALLOWED < 101300))
```

Comment:
This should probably be without leading underscores. Unless it uses `Availability.h` (which should not be done) directly instead of `AvailabilityMacros.h`.

Reply:
Also, older macOS SDKs don't define the `TARGET_OS_*` macros, and newer clang versions will error if you use them without checking if they are defined. So something like:
```
#if ((defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE && __IPHONE_OS_VERSION_MAX_ALLOWED < 110000) || ((!defined(TARGET_OS_OSX) || TARGET_OS_OSX) && MAC_OS_X_VERSION_MAX_ALLOWED < 101300))
```

Reply:
@jmroot - It looks like the brackets are not right in the second and third code segment:

Shouldn't it be:

`#if ((defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE && __IPHONE_OS_VERSION_MAX_ALLOWED < 110000) || (!defined(TARGET_OS_OSX)) || (TARGET_OS_OSX && MAC_OS_X_VERSION_MAX_ALLOWED < 101300))`

Reply:
The parens were indeed slightly off, and I've edited my previous comment to fix them. I think your version is equivalent because any version old enough to not define `TARGET_OS_OSX` is also less than 10.13.